community/kernel/src/test/java/org/neo4j/kernel/impl/storemigration/StoreMigratorTest.java

```diff
@@ -133,4 +146,39 @@ public void shouldBeAbleToResumeMigrationOnRebuildingCounts() throws Exception
                 new StoreFactory( fs, storeDirectory, pageCache, logService.getInternalLogProvider() );
         storeFactory.openNeoStoresEagerly().close();
     }
+
+    @Test
+    public void shouldComputeTheLastTxLogPositionCorrectly() throws Throwable
+    {
+        // GIVEN a legacy database
+        File storeDirectory = directory.graphDbDir();
+        File prepare = directory.directory( "prepare" );
+        MigrationTestUtils.prepareSampleLegacyDatabase( version, fs, storeDirectory, prepare );
+        // and a state of the migration saying that it has done the actual migration
+        LogService logService = NullLogService.getInstance();
+        PageCache pageCache = pageCacheRule.getPageCache( fs );
+        UpgradableDatabase upgradableDatabase =
+                new UpgradableDatabase( new StoreVersionCheck( pageCache ), new LegacyStoreVersionCheck( fs ) );
+
+        String versionToMigrateFrom = upgradableDatabase.checkUpgradeable( storeDirectory );
+        SilentMigrationProgressMonitor progressMonitor = new SilentMigrationProgressMonitor();
+        StoreMigrator migrator = new StoreMigrator( progressMonitor, fs, pageCache, new Config(), logService );
+        File migrationDir = new File( storeDirectory, StoreUpgrader.MIGRATION_DIRECTORY );
+        fs.mkdirs( migrationDir );
+
+        // WHEN migrating
+        migrator.migrate( storeDirectory, migrationDir, schemaIndexProvider, versionToMigrateFrom );
+
+        // THEN it should compute the correct last tx log position
+
+        try ( Reader reader = fs.openAsReader( new File( migrationDir, "lastxlogposition" ), Charsets.UTF_8.name() ) )
+        {
+            char[] buffer = new char[4096];
```

Comment:
Allocation of this array of size 4096 and following split are a bit mysterious. Maybe we can extract them into some helper method in `LogTestUtils`?